### PR TITLE
Return null in getCacheToUseIfEnabled when the protocol is not https

### PR DESCRIPTION
**Pull request by Bigtalljosh**
Addresses #23082 
If served over http the cache is assumed compromised, this can happen when the page first loads over http before the https redirect.

Weird error when running tests on my local machine: 
```
src/Platform/Profiling.ts:127:44 - error TS2304: Cannot find name 'BINDING'.
127     return typeof str === 'string' ? str : BINDING.conv_string(str);
```

After some digging looks like this is some Mono interface declared in `MonoTypes.ts`.  I'm not sure how to resolve this?

### CODE REVIEW

**Comment by Bigtalljosh** [comment]
I'm being a bit cheeky/lazy here and seeing if the ci builds pass to see if I have a local platform issue with that test error.

File: src/Components/Web.JS/src/Platform/WebAssemblyResourceLoader.ts [hunk 1]
```diff
@@ -165,6 +165,12 @@ async function getCacheToUseIfEnabled(bootConfig: BootJsonData): Promise<Cache |
     return null;
   }
 
+  // cache integrity is compromised if the first request has been served over http
+  // in this case, we want to disable caching and integrity validation
+  if (document.location.protocol === 'http:') {
```
Comment (pranavkm) [inline src/Components/Web.JS/src/Platform/WebAssemblyResourceLoader.ts:170]: Would it be better to require https instead? For e.g.

```suggestion
  if (document.location.protocol !== 'https:') {
```

Comment (Bigtalljosh) [reply src/Components/Web.JS/src/Platform/WebAssemblyResourceLoader.ts:170]: You might be right, specifically requiring https opposed to saying anything that isn't http. I'll change that now.

**Comment by Bigtalljosh** [comment]
Looking at my build failure there, am I supposed to be checking in `blazor.webassembly.js` ?
Sorry for the dumb question, this really is out my comfort zone I'm learning a lot today but there's just _so much_ I'm not used to. it's a lot to take in!

**Comment by javiercn** [comment]
> Looking at my build failure there, am I supposed to be checking in `blazor.webassembly.js` ?
> Sorry for the dumb question, this really is out my comfort zone I'm learning a lot today but there's just _so much_ I'm not used to. it's a lot to take in!

Ah, yes. You have to regenerate blazor.server.js and blazor.webassembly.js (just run dotnet build on Web.JS) this is due to one of our weird build constraints

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/dotnet/aspnetcore?pullRequest=24012) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/dotnet/aspnetcore?pullRequest=24012) before we can accept your contribution.<br/><br/>:x: Bigtalljosh [sign now](https://cla.dotnetfoundation.org/dotnet/aspnetcore?pullRequest=24012)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/dotnet/aspnetcore?pullRequest=24012) it.</sub>

**Review by pranavkm — APPROVED** [review]

**Comment by pranavkm** [comment]
Thanks for the PR!

**Pull request merged**